test/e2e/broker_trigger_test.go

```diff
@@ -0,0 +1,250 @@
+	if err := WaitForAllTriggersReady(clients, logger, ns); err != nil {
+		t.Fatalf("Error waiting for triggers to become ready: %v", err)
+	}
+
+	logger.Info("Triggers ready")
+
+	// These are the event types and sources that will be send.
+	eventsToSend := []test.TypeAndSource{
+		{eventType1, eventSource1},
+		{eventType1, eventSource2},
+		{eventType2, eventSource1},
+		{eventType2, eventSource2},
+	}
+
+	// We notice some crashLoopBacks in the filter and ingress pod creation.
+	// We then delay the creation of the sender pods in order not to miss events.
+	// TODO improve this
+	logger.Info("Waiting for filter and ingress pods to become running")
+	time.Sleep(waitForFilterPodRunning)
+
+	logger.Info("Creating event sender pods")
+
+	// Map to save the expected events per dumper so that we can verify the delivery.
+	expectedEvents := make(map[string][]string)
+	for _, eventToSend := range eventsToSend {
+		// Create cloud event.
+		// Using event type and source as part of the body for easier debugging.
+		body := fmt.Sprintf("Body-%s-%s", eventToSend.Type, eventToSend.Source)
+		cloudEvent := test.CloudEvent{
+			Source: eventToSend.Source,
+			Type:   eventToSend.Type,
+			Data:   fmt.Sprintf(`{"msg":%q}`, body),
+		}
+		// Create sender pod.
+		senderPodName := name("sender", eventToSend.Type, eventToSend.Source)
+		senderPod := test.EventSenderPod(senderPodName, ns, defaultBrokerUrl, cloudEvent)
+		if err := CreatePod(clients, senderPod, logger, cleaner); err != nil {
+			t.Fatalf("Error creating event sender pod: %v", err)
+		}
+
+		// Check on every dumper whether we should expect this event or not, and add its body
+		// to the expectedEvents map if so.
+		for _, eventToReceive := range eventsToReceive {
+			if shouldExpectEvent(&eventToSend, &eventToReceive, logger) {
+				subscriberPodName := name("dumper", eventToReceive.Type, eventToReceive.Source)
+				expectedEvents[subscriberPodName] = append(expectedEvents[subscriberPodName], body)
+			}
+		}
+	}
+
+	logger.Info("Event sender pods created")
+
+	logger.Info("Waiting for event sender pods to be running")
+
+	// Wait for all of them to be running.
+	if err := WaitForAllPodsRunning(clients, logger, ns); err != nil {
+		t.Fatalf("Error waiting for event sender pod to become running: %v", err)
+	}
+
+	logger.Info("Event sender pods running")
+
+	logger.Info("Verifying events delivered to appropriate dumpers")
+
+	for _, event := range eventsToReceive {
+		subscriberPodName := name("dumper", event.Type, event.Source)
+		subscriberPod := subscriberPods[subscriberPodName]
+		logger.Infof("Dumper %q expecting %q", subscriberPodName, strings.Join(expectedEvents[subscriberPodName], ","))
+		if err := WaitForLogContents(clients, logger, subscriberPodName, subscriberPod.Spec.Containers[0].Name, ns, expectedEvents[subscriberPodName]); err != nil {
+			t.Fatalf("String(s) not found in logs of subscriber pod %q: %v", subscriberPodName, err)
+		}
+	}
+
+	logger.Info("Verification successful!")
+
```

Comment:
Remove.

Reply:
Done

Reply:
I don't mind the log message being there or not (up to you), I just wanted the empty line before `}` removed.